test/basic.js

```diff
@@ -157,13 +157,17 @@ test('unsubscribe without subscribe', function (t) {
 })
 
 test('unsubscribe on disconnect for a clean=true client', function (t) {
-  t.plan(6)
+  t.plan(7)
 
   var opts = { clean: true }
-  var s = noError(connect(setup(), opts), t)
+  // var s = noError(connect(setup(), opts), t)
+  var s = connect(setup(), opts)
 
   subscribe(t, s, 'hello', 0, function () {
-    s.conn.emit('close')
+    // s.conn.emit('close')
+    s.conn.destroy(null, function () {
+      t.pass('closed streams')
+    })
```

Comment:
I've been trying to resolve the failing tests and it led me to think that s.conn.emit('close') was not closing streams properly. By replacing this event by the destroy method ( which will also emit the same event after ensuring streams are destroyed ), it seems that no data is received by the writeStream, but it triggered a clientError from the broker which in turns kills the test execution due to noError. So i removed it and the the 2 tests concerned are passing.
Is it an acceptable fix ?

Reply:
this is ok!